evm/vm/message.py

```diff
@@ -178,6 +180,32 @@ def __init__(self,
         validate_is_boolean(is_create, title="Message.is_create")
         self.is_create = is_create
 
+        validate_sig_hash(sig_hash, title="Message.sig_hash")
+        self.sig_hash = sig_hash
+
         if access_list is not None:
             validate_access_list(access_list)
         self.access_list = access_list
+
+    def prepare_child_message(self,
+                              gas,
+                              to,
+                              value,
+                              data,
+                              code,
+                              **kwargs):
+        kwargs.setdefault('sender', self.msg.storage_address)
+
+        child_message = ShardingMessage(
+            gas=gas,
+            gas_price=self.msg.gas_price,
+            origin=self.msg.origin,
+            sig_hash=self.msg.sig_hash,
```

Comment:
This has a "bad smell".

The `sighash` falls into the same category as the block data which indicates there should be a single location this value is stored.  Yet it is copied for each level of computation.

After some investigation, the same is true for `msg.gas_price` and `msg.origin`.  This isn't something we need to fix now, but I think it's worth investigating how we can remove this duplication. 

Issue opened here: https://github.com/ethereum/py-evm/issues/262

Reply:
I agree. I must admit that I simply copied the `msg.origin` behavior.